services/services.go

```diff
@@ -756,7 +797,7 @@ func getIndexPageData() (*types.IndexPageData, error) {
 		return nil, fmt.Errorf("error retrieving validator queue count: %v", err)
 	}
 	data.EnteringValidators = queueCount.EnteringValidators
-	data.ExitingValidators = queueCount.ExitingValidators
+	data.ExitingValidators = queueCount.ExitingValidators */
 
 	var epochLowerBound uint64
 	if epochLowerBound = 0; epoch > 1600 {
```

Comment:
I know this has not been touched by your changes but one could see it in the scope of this task, so:

Why are 1600 epochs checked here? Only the last finalized epoch and the generally last epoch are used. Wouldn't it suffice to, for example, only grab the last 10 epochs and if it does not contain a finalized one only THEN grab the last, let's say, 1600?

I have made a quick benchmark an the duration for the complete `epochHistory` handling would drop from ~950ms to ~220ms.

Thoughts?

Reply:
You're right, 1600 epochs is completely overkill. I've implemented changes so that the relevant db columns are already selected and evaluated in the first epoch query and left the original epoch history code as fallback if no finalized slot is found within the first queried epochs.

Reply:
Love it!